Proper TReflectionGameTab.Open function added & Widget left-click uptext check

NicklasSoderberg opened the pull request:
Proper TReflectionGameTab.Open function added & Widget left-click uptext check
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/54%23discussion_r26055676%22%2C%20%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/54%23discussion_r26055741%22%2C%20%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/54%23discussion_r26055800%22%2C%20%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/54%23discussion_r26055821%22%2C%20%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/54%23discussion_r26055839%22%2C%20%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/54%23discussion_r26055862%22%2C%20%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/54%23discussion_r26055897%22%2C%20%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/54%23discussion_r26055907%22%2C%20%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/54%23discussion_r26055919%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%20beaee7ce165b0d18778d00a6ef5f225add703a11%20lib/internal/Login.simba%20173%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/54%23discussion_r26055800%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22GetColor%5Cr%5Cn%22%2C%20%22created_at%22%3A%20%222015-03-09T17%3A22%3A21Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/4069146%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Elfyyy%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/internal/Login.simba%3AL234-248%22%7D%2C%20%22Pull%20beaee7ce165b0d18778d00a6ef5f225add703a11%20lib/internal/Login.simba%2073%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/54%23discussion_r26055676%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Mouse_Left%22%2C%20%22created_at%22%3A%20%222015-03-09T17%3A21%3A03Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/4069146%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Elfyyy%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/internal/Login.simba%3AL25-94%22%7D%2C%20%22Pull%20beaee7ce165b0d18778d00a6ef5f225add703a11%20lib/widgets/gametabs/Gametabs.simba%2025%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/54%23discussion_r26055919%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%5E%22%2C%20%22created_at%22%3A%20%222015-03-09T17%3A23%3A36Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/4069146%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Elfyyy%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/widgets/gametabs/Gametabs.simba%3AL30-61%22%7D%2C%20%22Pull%20beaee7ce165b0d18778d00a6ef5f225add703a11%20lib/internal/Login.simba%20199%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/54%23discussion_r26055839%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22not%2C%20Login_%22%2C%20%22created_at%22%3A%20%222015-03-09T17%3A22%3A45Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/4069146%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Elfyyy%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/internal/Login.simba%3AL252-268%22%7D%2C%20%22Pull%20beaee7ce165b0d18778d00a6ef5f225add703a11%20lib/internal/Login.simba%20191%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/54%23discussion_r26055821%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22GetColor%22%2C%20%22created_at%22%3A%20%222015-03-09T17%3A22%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/4069146%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Elfyyy%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/internal/Login.simba%3AL252-268%22%7D%2C%20%22Pull%20beaee7ce165b0d18778d00a6ef5f225add703a11%20lib/widgets/gametabs/Gametabs.simba%2016%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/54%23discussion_r26055907%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Boolean%5Cr%5Cn%22%2C%20%22created_at%22%3A%20%222015-03-09T17%3A23%3A31Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/4069146%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Elfyyy%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/widgets/gametabs/Gametabs.simba%3AL9-23%22%7D%2C%20%22Pull%20beaee7ce165b0d18778d00a6ef5f225add703a11%20lib/internal/Login.simba%20111%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/54%23discussion_r26055741%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Integer%5Cr%5Cn%22%2C%20%22created_at%22%3A%20%222015-03-09T17%3A21%3A43Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/4069146%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Elfyyy%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/internal/Login.simba%3AL25-94%22%7D%2C%20%22Pull%20beaee7ce165b0d18778d00a6ef5f225add703a11%20lib/widgets/Bank.simba%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/54%23discussion_r26055862%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Boolean%22%2C%20%22created_at%22%3A%20%222015-03-09T17%3A23%3A00Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/4069146%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Elfyyy%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/widgets/Bank.simba%3AL280-308%22%7D%2C%20%22Pull%20beaee7ce165b0d18778d00a6ef5f225add703a11%20lib/widgets/gametabs/Gametabs.simba%207%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/54%23discussion_r26055897%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22No%20need%20to%20reorder%20these%22%2C%20%22created_at%22%3A%20%222015-03-09T17%3A23%3A22Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/4069146%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Elfyyy%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/widgets/gametabs/Gametabs.simba%3AL9-23%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'><img src='http://www.codereviewhub.com/site/github-bar.png' height=40></a>
- [ ] <a href='#crh-comment-Pull beaee7ce165b0d18778d00a6ef5f225add703a11 lib/internal/Login.simba 73'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/Elfyyy/OSR-Reflection/pull/54#discussion_r26055676'>File: lib/internal/Login.simba:L25-94</a></b>
- <a href='https://github.com/Elfyyy'><img border=0 src='https://avatars.githubusercontent.com/u/4069146?v=3' height=16 width=16'></a> Mouse_Left
- [ ] <a href='#crh-comment-Pull beaee7ce165b0d18778d00a6ef5f225add703a11 lib/internal/Login.simba 111'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/Elfyyy/OSR-Reflection/pull/54#discussion_r26055741'>File: lib/internal/Login.simba:L25-94</a></b>
- <a href='https://github.com/Elfyyy'><img border=0 src='https://avatars.githubusercontent.com/u/4069146?v=3' height=16 width=16'></a> Integer
- [ ] <a href='#crh-comment-Pull beaee7ce165b0d18778d00a6ef5f225add703a11 lib/internal/Login.simba 173'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/Elfyyy/OSR-Reflection/pull/54#discussion_r26055800'>File: lib/internal/Login.simba:L234-248</a></b>
- <a href='https://github.com/Elfyyy'><img border=0 src='https://avatars.githubusercontent.com/u/4069146?v=3' height=16 width=16'></a> GetColor
- [ ] <a href='#crh-comment-Pull beaee7ce165b0d18778d00a6ef5f225add703a11 lib/internal/Login.simba 191'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/Elfyyy/OSR-Reflection/pull/54#discussion_r26055821'>File: lib/internal/Login.simba:L252-268</a></b>
- <a href='https://github.com/Elfyyy'><img border=0 src='https://avatars.githubusercontent.com/u/4069146?v=3' height=16 width=16'></a> GetColor
- [ ] <a href='#crh-comment-Pull beaee7ce165b0d18778d00a6ef5f225add703a11 lib/internal/Login.simba 199'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/Elfyyy/OSR-Reflection/pull/54#discussion_r26055839'>File: lib/internal/Login.simba:L252-268</a></b>
- <a href='https://github.com/Elfyyy'><img border=0 src='https://avatars.githubusercontent.com/u/4069146?v=3' height=16 width=16'></a> not, Login_
- [ ] <a href='#crh-comment-Pull beaee7ce165b0d18778d00a6ef5f225add703a11 lib/widgets/Bank.simba 5'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/Elfyyy/OSR-Reflection/pull/54#discussion_r26055862'>File: lib/widgets/Bank.simba:L280-308</a></b>
- <a href='https://github.com/Elfyyy'><img border=0 src='https://avatars.githubusercontent.com/u/4069146?v=3' height=16 width=16'></a> Boolean
- [ ] <a href='#crh-comment-Pull beaee7ce165b0d18778d00a6ef5f225add703a11 lib/widgets/gametabs/Gametabs.simba 7'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/Elfyyy/OSR-Reflection/pull/54#discussion_r26055897'>File: lib/widgets/gametabs/Gametabs.simba:L9-23</a></b>
- <a href='https://github.com/Elfyyy'><img border=0 src='https://avatars.githubusercontent.com/u/4069146?v=3' height=16 width=16'></a> No need to reorder these
- [ ] <a href='#crh-comment-Pull beaee7ce165b0d18778d00a6ef5f225add703a11 lib/widgets/gametabs/Gametabs.simba 16'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/Elfyyy/OSR-Reflection/pull/54#discussion_r26055907'>File: lib/widgets/gametabs/Gametabs.simba:L9-23</a></b>
- <a href='https://github.com/Elfyyy'><img border=0 src='https://avatars.githubusercontent.com/u/4069146?v=3' height=16 width=16'></a> Boolean
- [ ] <a href='#crh-comment-Pull beaee7ce165b0d18778d00a6ef5f225add703a11 lib/widgets/gametabs/Gametabs.simba 25'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/Elfyyy/OSR-Reflection/pull/54#discussion_r26055919'>File: lib/widgets/gametabs/Gametabs.simba:L30-61</a></b>
- <a href='https://github.com/Elfyyy'><img border=0 src='https://avatars.githubusercontent.com/u/4069146?v=3' height=16 width=16'></a> ^

<a href='https://www.codereviewhub.com/Elfyyy/OSR-Reflection/pull/54?mark_as_completed=1'><img src='http://www.codereviewhub.com/site/github-mark-as-completed.png' height=26></a>&nbsp;<a href='https://www.codereviewhub.com/Elfyyy/OSR-Reflection/pull/54?approve=1'><img src='http://www.codereviewhub.com/site/github-approve.png' height=26></a>&nbsp;<a href='https://github.com/Elfyyy/OSR-Reflection/pull/54'><img src='http://www.codereviewhub.com/site/github-refresh.png' height=26></a>
<a href='#crh-end'></a>

NicklasSoderberg committed:
Added left-click ability to widgets with uptext check

Added left-click ability to widgets with uptext check

NicklasSoderberg committed:
Proper TReflectionGameTab.Open function

Turned into a function, proper checks

Elfyyy commented:
This pull request is now managed with CodeReviewHub. <a href='https://github.com/Elfyyy/OSR-Reflection/pull/54'>Refresh</a>.<a href='#crh-update-none'></a>

NicklasSoderberg committed:
Added spaces because my standards are off

Uncle elfyyy sure told me.

NicklasSoderberg committed:
Fixed a small bug in Widgets & Fixed TReflectionBank.Close

Fixed a small bug in Widgets & Fixed TReflectionBank.Close

NicklasSoderberg committed:
Last commit before selective worlds before start.

Switched up a few things...
Added a proper isLoggedIn for LocalPlayer.
Widget akward error fixed
Clarification fix in login.simba (will change loginstate to
GetLoginState next commit).

NicklasSoderberg committed:
Fixed login, LocalPlayer.Changeworld is now flawless.

Timers timers and more timers

NicklasSoderberg committed:
Added custom start world to SMART

World := 318 before Reflect.Setup and you got yourself a :+1:

NicklasSoderberg committed:
Re-Added GetPage to GetWorld

Re-Added GetPage to GetWorld

NicklasSoderberg committed:
Standard fixes

Happy elf

KyleHunter commented on lib/internal/Login.simba:
```diff
@@ -48,50 +25,69 @@ begin
+    If not Reflect.Login.AtLogin then
+      Exit(False);
+    Inc(Tries);
+    Reflect.Mouse.Move(ClickSwitchButton, mouse_Left);
```
Mouse_Left

KyleHunter commented on lib/internal/Login.simba:
```diff
@@ -48,50 +25,69 @@ begin
 end;
 
-function TReflectionLogin.LoginState: integer;
+function TReflectionLogin.GetLoginState: integer;
```
Integer

KyleHunter commented on lib/internal/Login.simba:
```diff
@@ -205,14 +234,14 @@ var
   T, TLogged: TReflectTimer;
 begin
-  If Reflect.Login.GetState = 30 then
+  If (Reflect.Login.GetClientState = 30) and (getColor(427, 489) = 16777215) then
```
GetColor

KyleHunter commented on lib/internal/Login.simba:
```diff
@@ -223,15 +252,16 @@ begin
         TLogged.Start;
         repeat
-          Result := (Reflect.Login.LoginState = -1);
+          Result := (Reflect.Login.GetClientState = 30) and (getColor(427, 489) = 16777215);
```
GetColor

KyleHunter commented on lib/internal/Login.simba:
```diff
@@ -223,15 +252,16 @@ begin
       else
         Wait(RandomRange(100, 200));
     end;
     If Not (InIntArray([LOGIN_Connecting, LOGIN_ClickToPlay], CurState))then
```
not, Login_

KyleHunter commented on lib/widgets/Bank.simba:
```diff
@@ -280,15 +280,28 @@ begin
 end;
 
-procedure TReflectionBank.Close;
+function TReflectionBank.Close: boolean;
```
Boolean

KyleHunter commented on lib/widgets/gametabs/Gametabs.simba:
```diff
@@ -9,14 +9,14 @@ begin
-    Widget_Stats_Container, Widget_QuestList_Container,
-    Widget_Inventory_Container, Widget_WornEquipment_Container,
-    Widget_Prayer_Container, Widget_Magic_Container
+    Widget_Stats_Container, Widget_Inventory_Container,
```
No need to reorder these

KyleHunter commented on lib/widgets/gametabs/Gametabs.simba:
```diff
@@ -9,14 +9,14 @@ begin
-    Gametab_CombatOptions, Gametab_Stats, Gametab_QuestList, Gametab_Inventory,
-    Gametab_WornEquipment, Gametab_Prayer, Gametab_Magic
+    Gametab_CombatOptions, Gametab_Stats, Gametab_Inventory,
+    Gametab_WornEquipment, Gametab_Prayer, Gametab_Magic, Gametab_QuestList
```
Boolean

KyleHunter commented on lib/widgets/gametabs/Gametabs.simba:
```diff
@@ -30,14 +30,31 @@ begin
 end;
 
-procedure TReflectionGametab.Open(TabNum: Integer);
+function TReflectionGametab.Open(TabNum: Integer): boolean;
```
^

NicklasSoderberg committed:
Standards fix

Standards fix
